coalib/bearlib/aspects/meta.py

```diff
@@ -72,3 +73,21 @@ class Sub(subcls, aspectbase, metaclass=aspectclass):
 
     def __repr__(cls):
         return '<%s %s>' % (type(cls).__name__, repr(cls.__qualname__))
+
+
+def issubaspect(subaspect, aspect):
+    """
+    This function checks whether or not ``subaspect`` is a subaspect of
+    ``aspect``.
+    """
+    if (not isinstance(subaspect, aspectclass)
+            and not isinstance(subaspect, aspectbase)):
+        raise TypeError(
+            '{} is not an aspectclass or an instance of an '
+            'aspectclass'.format(repr(aspect)))
+    if (not isinstance(aspect, aspectclass)
+            and not isinstance(aspect, aspectbase)):
+        raise TypeError(
+            '{} is not an aspectclass or an instance of an '
+            'aspectclass'.format(repr(subaspect)))
+    return subaspect.__qualname__.startswith(aspect.__qualname__, 0, -1)
```

Comment:
There are 2 drawbacks here:

**[0]** One idea of `issubaspect` was that aspects are also considered as subaspects of themselves, just like `issubclass(str, str) is True`. The exclusion of the last character with stop index `-1` prevents this.

**[1]** Instances of aspectclasses don't have `__qualname__` anymore. You would need to use `type(subaspect).__qualname__` in that case.

**[2]** The `.startswith` approach creates false positives, just like the original `in` approach. Why didn't I realize that earlier? Shame on me! :D Just consider the following:

```python
@Root.subaspect
class SubAspect:
    """
    """

@Root.subaspect
class SubAspectYEAH:
    """
    """
```

```python
>>> issubaspect(SubAspectYEAH, SubAspect)
True
```

To solve **[2]**, it must be additionally checked if the `__qualname__` of the parent is followed by a dot in or the end of the subaspect's `__qualname__`. IMO the simplest way would be a regex here:

```python
    return bool(re.match(r'^%s(\.|$)' % aspect.__qualname__, subaspect.__qualname__))
```

Meanwhile I think the recursive `.parent` check approach would actually be simpler in the end ;) Since `.parent` is available on aspectclasses AND their instances.